### ISSUE TITLE
Jump to unread displays in some impropor circumstances

### ISSUE THREAD

**Issue by Autre31415** [vector-im/riot-web#4180]
I see this message in chats pop up every so often even though i've seen all the messages already 

![unread message](https://cloud.githubusercontent.com/assets/8182831/26757297/a5cdcb3c-4885-11e7-809e-a6e9417ff798.png)

OS: macOS Sierra
riot-web: 0.10.1

**Comment by lukebarnard1** [comment]
@Autre31415 does the bar appear even when you've clicked the "X"?

**Comment by Autre31415** [comment]
The 'X' does get rid of it.

**Comment by lukebarnard1** [comment]
OK, then it sounds like you're expecting it to disappear when you scroll down but it doesn't because you need to scroll quite slowly. If you click the "Jump to.." and then scroll down quickly, would you expect the banner to disappear?

**Comment by Autre31415** [comment]
This isn't an issue that I've witnessed in prior versions of Riot.

The jump to unread indicator has a tendency to pop up while I'm actively engaged in a conversation, and jumping to the first unread message will take it all the way back to when the conversation started even though I've been active in the conversation the entire time and no notifications have spawned.

**Comment by lukebarnard1** [comment]
Right, so that sounds like a bug where the app's/view's read marker is not up-to-date with the server's.

**Comment by 4nd3r** [comment]
> Ander: ugh, when will the "Jump to first unread" will be fixed? :<
> Dave: we just did a bunch of fixes, if it's not working please file a bug
> Ander: just did? i'm using 0.11.4. or are those fixes in develop?
> Dave: they're in 0.11.4

so, with 0.11.4 it's still on. this annoying pop up comes up quite often. clicking on it takes me back more than a day or days, even if the last message in room was sent by me few minutes ago.

**Comment by lukebarnard1** [comment]
@4nd3r you can get rid of it by clicking "X". 

>clicking on it takes me back more than a day or days

This is expected behaviour

>even if the last message in room was sent by me few minutes ago

And so is this. If you are just finding the pop up annoying or unintuitive, see https://github.com/vector-im/riot-web/issues/4134

**Comment by pravi** [comment]
@lukebarnard1 I think I found the root cause. When more than 3 collapsed activities are present, you need to expand at least one of them for the read marker to consider it read. So even if you scroll down slowly, the read marker does not get updated and takes you far back to the first unread message every time since that cannot be marked as read unless you expand the collapsed messages. This is an issue in bridged rooms because IRC or Telegram bridges generate a lot of join/left or profile pic changed messages.

**Comment by TripleSnail** [comment]
I can almost never rely on "Jump to first unread message.". I have a simple 1-on-1 chat, where nobody leaves, changes names etc. Sometimes it takes me a week back. Half of the time it takes me back to **my own message** (it underlines my own message). It also takes me back to **redacted events**, closed a call events and so on.

**Comment by lukebarnard1** [comment]
@TripleSnail, none of that is unexpected behaviour afaik - in reality we track the first unread _event_. 

If you send a message into the room and then leave your client reading messages into the room (without focusing on the window) your read marker will still be underneath the message that you sent - it was the last message you read.

**Comment by TripleSnail** [comment]
I had always understood, that it underlined the message that I had not read. Well, I guess everything makes more sense now.

**Comment by BloodyIron** [comment]
I see this message ALL THE TIME when I'm already current in a channel. Can we please get this back into the pipeline in some way?
